core/src/main/java/org/apache/brooklyn/feed/ssh/SshFeed.java

```diff
@@ -270,6 +270,7 @@ private SshPollValue exec(String command, Map<String,String> env) throws IOExcep
         ConfigBag flags = ConfigBag.newInstanceExtending(config().getBag())
             .configure(SshTool.PROP_NO_EXTRA_OUTPUT, true)
             .configure(SshTool.PROP_OUT_STREAM, stdout)
+            .configure(SshMachineLocation.CLOSE_CONNECTION, true)
```

Comment:
I say "really expensive" because establishing an ssh connection is cpu-intensive. By caching the connection and re-using it, we cut that cpu-usage dramatically.